src/SmiServices/Microservices/DicomAnonymiser/Anonymisers/SmiCtpAnonymiser.cs

```diff
@@ -0,0 +1,112 @@
+using NLog;
+using SmiServices.Common.Messages.Extraction;
+using SmiServices.Common.Options;
+using SmiServices.Microservices.DicomAnonymiser.Helpers;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Abstractions;
+using System.Threading;
+
+namespace SmiServices.Microservices.DicomAnonymiser.Anonymisers;
+
+public class SmiCtpAnonymiser : IDicomAnonymiser, IDisposable
+{
+    private readonly ILogger _logger = LogManager.GetCurrentClassLogger();
+    private readonly Process _ctpProcess;
+
+    // NOTE(rkm 2025-01-08) This sometimes takes more than 10s in CI for some reason
+    private readonly TimeSpan CTP_TIMEOUT = TimeSpan.FromSeconds(20);
+
+    public SmiCtpAnonymiser(GlobalOptions globalOptions)
+    {
+        var dicomAnonymiserOptions = globalOptions.DicomAnonymiserOptions ?? throw new ArgumentException($"{nameof(globalOptions.DicomAnonymiserOptions)} was null", nameof(globalOptions));
+
+        if (!File.Exists(dicomAnonymiserOptions.CtpAnonCliJar))
+            throw new ArgumentException($"{nameof(dicomAnonymiserOptions.CtpAnonCliJar)} '{dicomAnonymiserOptions.CtpAnonCliJar}' does not exist", nameof(globalOptions));
+
+        if (!File.Exists(dicomAnonymiserOptions.CtpAllowlistScript))
+            throw new ArgumentException($"{nameof(dicomAnonymiserOptions.CtpAllowlistScript)} '{dicomAnonymiserOptions.CtpAllowlistScript}' does not exist", nameof(globalOptions));
+
+        var srAnonTool = "false";
+        if (!string.IsNullOrWhiteSpace(dicomAnonymiserOptions.SRAnonymiserToolPath))
+        {
+            if (!File.Exists(dicomAnonymiserOptions.SRAnonymiserToolPath))
+                throw new ArgumentException($"{nameof(dicomAnonymiserOptions.SRAnonymiserToolPath)} '{dicomAnonymiserOptions.SRAnonymiserToolPath}' does not exist", nameof(globalOptions));
+            srAnonTool = dicomAnonymiserOptions.SRAnonymiserToolPath;
+        }
+
+        var ctpArgs = $"-jar {dicomAnonymiserOptions.CtpAnonCliJar} --anon-script {dicomAnonymiserOptions.CtpAllowlistScript} --sr-anon-tool {srAnonTool} --daemonize";
+        _logger.Info($"Starting ctp with: java {ctpArgs}");
+
+        var ready = false;
+        _ctpProcess = ProcessWrapper.CreateProcess("java", ctpArgs);
+        _ctpProcess.OutputDataReceived += OnCtpOutputDataReceived;
+        _ctpProcess.ErrorDataReceived += (_, args) => _logger.Debug(args.Data);
+        _ctpProcess.Start();
+        _ctpProcess.BeginOutputReadLine();
+        _ctpProcess.BeginErrorReadLine();
+
+        lock (_ctpProcess)
+            Monitor.Wait(_ctpProcess, CTP_TIMEOUT);
+
```

Comment:
hm, deliberately leaving the first STDOUT handler in place so it logs everything we get back without duplicating the logging code in the second phase event handler?

Reply:
That was the intention, yeah!